src/main/scala/treadle/executable/ExpressionCompiler.scala

```diff
@@ -32,6 +34,19 @@ class ExpressionCompiler(
     }
   }
 
+  // Preset registers have references to the async reset
+  private val presetTargetList = presetTargets.map { presetTarget =>
+    val pathSteps = mutable.ArrayBuffer[String]()
+    var element = presetTarget.path
+    while (element.nonEmpty) {
+      val step = element.head._1.value
+      pathSteps += step
+      element = element.tail
+    }
+    pathSteps += presetTarget.ref
+    pathSteps.mkString(".")
+  }
+
   def getWidth(expression: Expression): Int = {
```

Comment:
FYI, I think there is already a `firrtl.bitWidth` function that may do the same thing.